app/src/main/java/ar/edu/unlam/mobile/scaffolding/ui/screens/publicationEdit/PublicationEditViewModel.kt

```diff
@@ -103,4 +112,156 @@ class PublicationEditViewModel
+        val dateLost: State<String> = _dateLost
+
+        private val _species = mutableStateOf("")
+        val species: State<String> = _species
+
+        private val _sex = mutableStateOf("")
+        val sex: State<String> = _sex
+
+        private val _age = mutableStateOf("")
+        val age: State<String> = _age
+
+        private val _color = mutableStateOf("")
+        val color: State<String> = _color
+
+        private val _location = mutableStateOf("")
+        val location: State<String> = _location
+
+        private val _contact = mutableStateOf("")
+        val contact: State<String> = _contact
+
+        private val dateFormat = SimpleDateFormat("dd/MM/yyyy")
+
+        private fun setId(value: String) {
+            _id.value = value
+        }
+
+        fun setType(value: String) {
+            _type.value = value
+        }
+
+        fun setTitle(value: String) {
+            _title.value = value
+        }
+
+        fun setDescription(value: String) {
+            _description.value = value
+        }
+
+        fun setDateLost(value: String) {
+            val formattedDate: String =
+                try {
+                    dateFormat.parse(value)?.let { dateFormat.format(it) } ?: dateFormat.format(Date())
+                } catch (e: ParseException) {
+                    dateFormat.format(Date()) // Si hay un error al analizar, devuelve la fecha actual como una cadena
+                }
+            _dateLost.value = formattedDate
+        }
+
+        fun setSpecies(value: String) {
+            _species.value = value
+        }
+
+        fun setSex(value: String) {
+            _sex.value = value
+        }
+
+        fun setAge(value: String) {
+            _age.value = value
+        }
+
+        fun setColor(value: String) {
+            _color.value = value
+        }
+
+        fun setLocation(value: String) {
+            _location.value = value
+        }
+
+        fun setContact(value: String) {
+            _contact.value = value
+        }
+
+        fun newPublication() {
+            if (title.value !== "" &&
+                description.value !== "" &&
+                location.value !== "" &&
+                type.value !== "" &&
+                age.value !== "" &&
+                contact.value !== ""
+            ) {
+                setId(UUID.randomUUID().toString())
+
+                if (dateLost.value.isEmpty()) {
+                    setDateLost(dateFormat.format(Date())) // Devuelve la fecha actual como una cadena
+                }
+                if (species.value.isEmpty()) {
+                    setSpecies(Species.LORO.toString())
+                }
+                if (sex.value.isEmpty()) {
+                    setSex(Sex.MACHO.toString())
+                }
+                if (color.value.isEmpty()) {
+                    setColor(PetColors.MARRON.toString())
+                }
+                currentUserId?.let { addPublication(it.userId) }
```

Comment:
Podria estar dentro de una función que sea 
`fun validateForm(): Boolean {
    var isValid = true

    if (userName.isNullOrEmpty()) {
        userNameTextField.setError("Username is required")
        isValid = false
    }

    if (email.isNullOrEmpty()) {
        emailTextField.setError("Email is required")
        isValid = false
    }

    if (password.isNullOrEmpty()) {
        passwordTextField.setError("Password is required")
        isValid = false
    }

    // Añadir validación para otros campos según sea necesario

    return isValid
}`

O similar